### PR TITLE
Fix restore app startup and shutdown behavior

### LINKED ISSUE #10824: Restore app fails to inject backup store
Starting the restore app with all necessary configuration fails because Spring can't inject the backup store:

```
Error creating bean with name 'backupStore' defined in class path resource [io/camunda/zeebe/restore/BackupStoreComponent.class]:
Invalid destruction signature; nested exception is org.springframework.beans.factory.support.BeanDefinitionValidationException:
Could not find a destroy method named 'close' on bean with name 'backupStore'
```

### PR DESCRIPTION
**Pull request by lenaschoenburg**
- fixes injection of backup store
- fixes missing shutdown of restore app
- fixes a confusing and noisy exception when restore fails within a docker container

closes #10824

### CODE REVIEW

**Review by deepthidevaki — APPROVED** [review]
:rocket: Thanks for fixing it.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/RestoreAcceptanceIT.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        .untilAsserted(
+            () -> {
+              final var status = actuator.status(response.id());
+              assertThat(status)
+                  .extracting(BackupStatus::backupId, BackupStatus::status)
+                  .containsExactly(1L, BackupStatusCode.COMPLETED);
+            });
+
+    // then -- restore container exits with an error code
+    assertThatCode(
```
[the remark this answers is not in the document]
Comment (oleschoenburg) [reply qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/RestoreAcceptanceIT.java:155]: Ahh, I always forget about this :+1:

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
Of course..

bors merge

**Comment by oleschoenburg** [comment]
bors retry

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**